tests/vcs/VCSCommitBearTest.py

```diff
@@ -0,0 +1,189 @@
+
+    def test_head_commit(self):
+        self.assertEqual(self.run_uut(), ['HEAD commit information'])
+
+
+@generate_skip_decorator(GitCommitBear)
+class AnalyzeGitCommitTest(unittest.TestCase):
+
+    def run_uut(self, *args, **kwargs):
+        """
+        Runs the unit-under-test (via `self.uut.run()`) and collects the
+        messages of the yielded results as a list.
+
+        :param args:   Positional arguments to forward to the run function.
+        :param kwargs: Keyword arguments to forward to the run function.
+        :return:       A list of result values that give information related
+                       to head commit.
+        """
+        return list((result.raw_commit_message,
+                     result.commit_sha,
+                     result.commit_type,
+                     result.added_files,
+                     result.modified_files,
+                     result.deleted_files) for result in self.uut.run(
+                     *args, **kwargs))
+
+    def setUp(self):
+        self.msg_queue = Queue()
+        self.section = Section('')
+        self.uut = GitCommitBear(None, self.section, self.msg_queue)
+
+        self._old_cwd = os.getcwd()
+        self.gitdir = mkdtemp()
+        os.chdir(self.gitdir)
+        GitCommitBearTest.run_git_command('init')
+        GitCommitBearTest.run_git_command(
+            'config', 'user.email coala@coala.io')
+        GitCommitBearTest.run_git_command('config', 'user.name coala')
+
+    @staticmethod
+    def _windows_rmtree_remove_readonly(func, path, excinfo):
+        os.chmod(path, stat.S_IWRITE)
+        func(path)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        if platform.system() == 'Windows':
+            onerror = self._windows_rmtree_remove_readonly
+        else:
+            onerror = None
+        shutil.rmtree(self.gitdir, onerror=onerror)
+
+    def test_analyze_git_commit(self):
+        run_shell_command('git checkout -b testbranch master')
+        run_shell_command('touch testfile1.txt ')
+        run_shell_command('git add testfile1.txt')
+        run_shell_command('git commit -m "Add testfile1"')
+        test_sha1 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        self.assertEqual(self.run_uut(),
+                         [('Add testfile1\n\n', test_sha1, ['merge_commit'],
+                           ['testfile1.txt'], [], [])])
+
+        run_shell_command('touch testfile2.txt')
+        run_shell_command('git add testfile2.txt')
+        run_shell_command('git commit -m "Add testfile2 [ci skip]"')
+        test_sha2 = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+        self.assertEqual(self.run_uut(),
+                         [('Add testfile2 [ci skip]\n\n', test_sha2,
+                           ['ci_skip_commit', 'merge_commit'],
+                           ['testfile2.txt'], [], [])])
+
+        run_shell_command(['echo Some text here > testfile2.txt'], shell=True)
```

Comment:
do not use `echo` & `shell=True`.
Create the file using normal Python file routines.